### ISSUE TITLE
wrong error message for non-overlapping identity check with generics

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#11968]
```py
list[int] is list[str]
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-context%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Clocal-partial-types%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=738641794bba185f289f6e8faecd3983

# expected
> error: Non-overlapping identity check (left operand type: "Type[List[**int**]]", right operand type: "Type[List[**str**]]")

# actual
> error: Non-overlapping identity check (left operand type: "Type[List[**Any**]]", right operand type: "Type[List[**Any**]]")

**Comment by KotlinIsland** [comment]
```py
list[int] is list
```
This doesn't have any error.

**Comment by ilevkivskyi** [comment]
The error message now looks correct
```
error: Non-overlapping identity check (left operand type: "type[list[int]]", right operand type: "type[list[str]]")  [comparison-overlap]
```
